src/handle_action.cpp

```diff
@@ -730,21 +730,85 @@ static void haul()
     Character &player_character = get_player_character();
     map &here = get_map();
 
-    if( player_character.is_hauling() ) {
-        player_character.stop_hauling();
-    } else {
-        if( here.veh_at( player_character.pos() ) ) {
-            add_msg( m_info, _( "You cannot haul inside vehicles." ) );
-        } else if( here.has_flag( ter_furn_flag::TFLAG_DEEP_WATER, player_character.pos() ) ) {
-            add_msg( m_info, _( "You cannot haul while in deep water." ) );
-        } else if( !here.can_put_items( player_character.pos() ) ) {
-            add_msg( m_info, _( "You cannot haul items here." ) );
-        } else if( !here.has_haulable_items( player_character.pos() ) ) {
-            add_msg( m_info, _( "There are no items to haul here." ) );
-        } else {
-            player_character.start_hauling();
+    bool exit = false;
+    do {
+        uilist menu;
+
+        bool hauling = player_character.is_hauling();
+        bool selective = player_character.is_hauling_selectively();
+
+        std::string header = _( "You are currently " );
+
+        header += hauling ? _( "hauling items." ) : _( "not hauling items." );
```

Comment:
```suggestion
        std::string header = hauling ? _( "You are currently hauling items." ) : _( "You are currently not hauling items." );
```
Other languages do not necessarily have the same word order as English.